### PR TITLE
Avoid relaunching the analysis onboarding by interacting with a widget

### LINKED ISSUE #10316: onboarding dialog is shown everytime a widget to the source changes
STR:
1) add a points table
2) add a sampling analysis (10%)
3) add a wiget to the source table

everytime you filter the builder shows the onboarding dialog

#SMOKE_AN05

### PR DESCRIPTION
**Pull request by javierarce**
Closes #10316 

This PR fixes an issue where an analysis onboarding appeared after a widget attached to the layer with the analysis was changed.

The difficulty of solving this issue is that both widgets and user interactions can trigger the analysis [changed binding](https://github.com/CartoDB/cartodb/compare/10316-onboarding-reopen?expand=1#diff-3d74f8298f087839dd188fe931ea5322L195), and we can't really guess which of them did it by just looking to the analysis node definition. 

The solution I propose is to set a flag (`USER_SAVED`) to indicate that the user updated the analysis. This way, anytime we need to show the onboarding, we can check that this flag was set (I'm using a flag instead of setting a property to the model to avoid storing it permanently in the model).

Are you OK with this approach @xavijam & @javisantana?

### CODE REVIEW

**Comment by javisantana** [comment]
a question, when i reload the builder i will get the messages again, right?

**Comment by javierarce** [comment]
No, you shouldn't get it, since the flag won't be set then (it only gets set whenever you save the analysis again and it's not stored permanently).

**Comment by javierarce** [comment]
CR: @xavijam

**Comment by xavijam** [comment]
If we agree this decision (which is fine by me) I'd add a test .

**Comment by javierarce** [comment]
Pushed some specs. 🙏  CR: @xavijam

File: lib/assets/javascripts/cartodb3/data/user-actions.js [hunk 1]
```diff
@@ -178,6 +178,7 @@ module.exports = function (params) {
       var nodeDefModel = analysisDefinitionNodesCollection.get(analysisFormModel.id);
 
       if (nodeDefModel) {
+        nodeDefModel.USER_SAVED = true;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/user-actions.js]: Not sure if we should move this "local" variable to directly analysis-definition-model and set it in the initialize, what do you think @javierarce?

Comment (javierarce) [inline lib/assets/javascripts/cartodb3/data/user-actions.js]: Yes. You mean put it on the `initialize` method and set it `false` by default, right?

Comment (javierarce) [inline lib/assets/javascripts/cartodb3/data/user-actions.js]: Here's the change: https://github.com/CartoDB/cartodb/pull/10336/commits/6ed57e6bdca13ed4cdac1a0ab606a13de90af739

File: lib/assets/javascripts/cartodb3/data/analysis-definition-model.js [hunk 2]
```diff
@@ -16,6 +16,8 @@ module.exports = Backbone.Model.extend({
   initialize: function (attrs, opts) {
     if (!opts.analysisDefinitionNodesCollection) throw new Error('analysisDefinitionNodesCollection is required');
 
+    this.USER_SAVED = false;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/analysis-definition-model.js]: :+1:, could we add a description over the variable in order to know why we took that decision? thanks man and sorry to be picky.

**Comment by javierarce** [comment]
Ready to go?

**Comment by xavijam** [comment]
Fine by me

**Pull request merged**